src/plugins/ucx/ucx_backend.cpp

```diff
@@ -431,11 +451,20 @@ nixlUcxEngine::nixlUcxEngine (const nixlBackendInitParams* init_params)
     nixl_b_params_t* custom_params = init_params->customParams;
 
     if (init_params->enableProgTh) {
+        pthrOn = true;
         if (!nixlUcxContext::mtLevelIsSupproted(NIXL_UCX_MT_WORKER)) {
             NIXL_ERROR << "UCX library does not support multi-threading";
             this->initErr = true;
             return;
         }
+        pthrControlFd = eventfd(0, EFD_NONBLOCK);
```

Comment:
is event fd posix compliant? since posix compliancy argument justified using poll rather than epoll